edgedb/abstract.py

```diff
@@ -47,6 +71,19 @@ class ExecuteContext(typing.NamedTuple):
     state: typing.Optional[options.State]
 
 
+class DescribeContext(typing.NamedTuple):
```

Comment:
Now these has to be dataclasses, right? Since we don't need to support earlier python versions.

Reply:
Why? They are for internal use only and I think named tuples are slightly more efficient? Sorry I didn’t get your point.

Reply:
Ah, well, yes. DescribeContext is internal. But DescribeResult isn't (when `describe_query` becomes public).

Generally point is: `NamedTuple` are the old "poor man's dataclass". If there is no backwards compatibility concerns and no need to compare with tuple, it's better to use dataclass. For example, adding a field into dataclass in the future version is more expected (while for named tuple, different tuple length is technically a breaking change).

And I don't think performance matters here (I also believe that eventually dataclasses should be in par with namedtuples with regards to performance).